lib/file_transfer_agent/encrypt_util.js

```diff
@@ -132,6 +131,12 @@ function EncryptUtil(encrypt, filestream, temp) {
       keySize * 8
     );
 
+    if (keyIv && dataAad && keyAad) {
```

Comment:
Will it work for empty strings? What if `keyIv` is not empty, but `dataAad` is and we have ANDs?

Reply:
these are actually buffer objects and not strings. Still I refactored it to ternary operator encoding.